core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3368,8 +3396,16 @@ object KafkaApis {
   private[server] def sizeOfThrottledPartitions(versionId: Short,
                                                 unconvertedResponse: FetchResponse,
                                                 quota: ReplicationQuotaManager): Int = {
-    FetchResponse.sizeOf(versionId, unconvertedResponse.responseData.entrySet
-      .iterator.asScala.filter(element => quota.isThrottled(element.getKey)).asJava)
+    val topicResponses = new util.ArrayList[FetchResponseData.FetchableTopicResponse]()
```

Comment:
Do we have to create a new collection? The previous version did not.

Reply:
> The previous version did not.

It created copy before since we have to add a collection to auto-generated data to calculate the size. see https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java#L133

Reply:
I see, I guess this changed once we moved to the auto generated fetch.